pkg/controllers/allocation/filter.go

```diff
@@ -63,6 +57,35 @@ func (f *Filter) GetProvisionablePods(ctx context.Context, provisioner *v1alpha2
 	return provisionable, nil
 }
 
+// GetProvisionerFor retrieves the provisioner responsible for the pod
+func (f *Filter) GetProvisionerFor(ctx context.Context, p *v1.Pod) (*v1alpha2.Provisioner, error) {
+	provisionerKey := client.ObjectKey{
+		Namespace: "default",
+		Name:      "default",
+	}
+	if name, ok := p.Spec.NodeSelector[v1alpha2.ProvisionerNameLabelKey]; ok {
+		provisionerKey.Name = name
+	}
+	if namespace, ok := p.Spec.NodeSelector[v1alpha2.ProvisionerNamespaceLabelKey]; ok {
+		provisionerKey.Namespace = namespace
+	}
+	provisioner := &v1alpha2.Provisioner{}
+	if err := f.kubeClient.Get(ctx, provisionerKey, provisioner); err != nil {
+		return nil, err
```

Comment:
Can you make sure that this error message is as crisp as possible. We need to make it absolutely apparent that the default provisioner is required.

Reply:
Perhaps w/ a IsNotFound check

Reply:
The default provisioner is not required if every pod gets provisioner labels. Now I'm kind of thinking that's why you wanted the provisioner in-lined to make these two error messages distinct? Am I warm?

Reply:
not warm, but took a stab at a clear error message, lmk what you think

Reply:
We're saying different things. I'm thinking specifically of the cases:

1. If a provisioner doesn't exist, we should state, loudly, "provisioner does not exist, please specify karpenter.sh/provisioner-name ... or create a default provisioner in the default namespace" 
2. Auth error, use normal error handling logic.